specification/loadtestservice/LoadTestService/models.tsp

```diff
@@ -283,40 +347,127 @@ not required to upload.
 
   @doc("Optional load test config")
   optionalLoadTestConfig?: OptionalLoadTestConfig;
+
+  @added(APIVersions.v2023_04_01_preview)
+  @doc("Type of test. e.g. QuickTest, UrlBasedTest")
+  testType?: TestType;
+}
+
+@doc("Quick test config")
+model QuickTestConfig {
+  @doc("Test URL. Provide the complete HTTP URL. For example, http://contoso-app.azurewebsites.net/login")
+  endpointUrl?: string;
+
+  @added(APIVersions.v2023_04_01_preview)
+  @doc("Throughput required (requests per second)")
+  rps?: int32;        
+
+  @added(APIVersions.v2023_04_01_preview)
+  @doc("Maximum response time of the API/endpoint")
+  maxResponseTime?: int32;
+}
+
+@doc("Url test config")
+model UrlTestConfig {
+  @added(APIVersions.v2023_04_01_preview)
+  @doc("Number of steps in the ramp - up phase")
+  rampUpSteps?: int32;  
+
+  @added(APIVersions.v2023_04_01_preview)
+  @doc("Time to maintain a higher load after a spike in traffic")
+  spikeHoldLoad?: int32;
+
+  @added(APIVersions.v2023_04_01_preview)
+  @doc("Multiplier applied to the base load during a spike in traffic")
+  spikeMultiplier?: float32;
+
+  @added(APIVersions.v2023_04_01_preview)
+  @doc("Type of load. e.g. Linear, Constant, StepUp, Spike")
+  loadType?: LoadType;    
```

Comment:
loadType is an optional parameter should we make it mandatory?

Reply:
So this all depends on what we decide on Polymorphism question. Currently we have all fields inside OptionalLoadConfig and there loadType would be optional.